fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/recordrepair/RecordValidateAndRepairTest.java

```diff
@@ -46,125 +48,59 @@
-        final RecordMetaDataHook hook = ValidationTestUtils.getRecordMetaDataHook(splitLongRecords, storeVersions);
-        List<FDBStoredRecord<Message>> savedRecords = saveRecords(splitLongRecords, formatVersion, hook);
-        // Delete a record
-        try (FDBRecordContext context = openContext()) {
-            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
-            // Note that the primary keys start with 1, so the location is one-off when removed
-            store.deleteRecord(savedRecords.get(ValidationTestUtils.RECORD_INDEX_WITH_NO_SPLITS).getPrimaryKey());
-            store.deleteRecord(savedRecords.get(ValidationTestUtils.RECORD_INDEX_WITH_THREE_SPLITS).getPrimaryKey());
-            store.deleteRecord(savedRecords.get(21).getPrimaryKey());
-            store.deleteRecord(savedRecords.get(22).getPrimaryKey());
-            store.deleteRecord(savedRecords.get(44).getPrimaryKey());
-            commit(context);
-        }
-
-        List<RecordValidationResult> repairResults;
-
-        try (FDBRecordContext context = openContext()) {
-            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
-            RecordRepairRunner runner = RecordRepairRunner.builder(fdb).build();
-            repairResults = runner.runValidationAndRepair(store.asBuilder(), validationKind, true);
-        }
-
-        // Verify records: The missing records are gone, so won't be flagged, leaving only 45 records around.
-        if (storeVersions || validationKind.equals(RecordRepairRunner.ValidationKind.RECORD_VALUE)) {
-            Assertions.assertThat(repairResults).hasSize(0);
-        } else {
-            Assertions.assertThat(repairResults).hasSize(45);
-            Assertions.assertThat(repairResults).allMatch(result ->
-                    (!result.isValid()) && result.getErrorCode().equals(RecordVersionValidator.CODE_VERSION_MISSING_ERROR) &&
-                            result.isRepaired() &&
-                            result.getRepairCode().equals(RecordVersionValidator.REPAIR_VERSION_CREATED)
-            );
+        RecordRepair.Builder builder = RecordRepair.builder(fdb, storeBuilder).withValidationKind(validationKind);
+        // Run validation and repair
+        try (RecordRepairValidateRunner runner = builder.buildRepairRunner(true)) {
+            RepairValidationResults repairResults = runner.run().join();
+            ValidationTestUtils.assertCompleteResults(repairResults, NUM_RECORDS);
+            // Verify records: all is OK.
+            // If we are storing versions, they will all be there
+            // If we are not storing versions, verifying them is a no-op, so none will be flagged
+            ValidationTestUtils.assertInvalidResults(repairResults.getInvalidResults(), 0, null);
```

Comment:
_perhaps_  it would be better to just do:
```suggestion
            assertThat(repairResults.getInvalidResults()).isEmpty();
```

In all the cases where it's expected to be entirely valid. The `null` is a bit confusing, and this seems simpler than adding a:
```
ValidationTestUtils.assertNoInvalidResults(repairResults.getInvalidResults());
```